GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs

```diff
@@ -231,25 +238,21 @@ public void Recall()
         public bool LevelUp()
         {
             var stats = Stats;
-            var expMap = _game.Map.MapProperties.ExpToLevelUp;
-            if (stats.Level >= expMap.Count)
-            {
-                return false;
-            }
-
-            if (stats.Experience < expMap[stats.Level])
-            {
-                return false;
-            }
+            var expMap = _game.Config.MapData.ExpCurve;
 
-            while (stats.Level < expMap.Count && stats.Experience >= expMap[stats.Level])
+            //Ideally we'd use "stats.Level < expMap.Count + 1", but since we still don't have gamemodes implemented yet, i'll be hardcoding the EXP level to cap at lvl 18,
+            //Since the SR Map has 30 levels in total because of URF
+            while (stats.Level < 18 && (stats.Level < 1 || stats.Experience >= expMap[stats.Level - 1])) //The + and - 1s are there because the XP files don't have level 1
```

Comment:
Why this loop is while if you return true inside immediately? Maybe change to `if`?